lib/src/widgets/credit_card_button.dart

```diff
@@ -0,0 +1,107 @@
+import 'package:flutter/material.dart';
+
+import 'package:moyasar/moyasar.dart';
+import 'package:moyasar/src/moyasar.dart';
+
+import 'package:moyasar/src/models/payment_request.dart';
+import 'package:moyasar/src/models/sources/card/card_request_source.dart';
+
+import 'package:moyasar/src/widgets/three_d_s_webview.dart';
+
+/// The widget that shows the Credit Card form and manages the 3DS step.
+class CreditCardButton extends StatefulWidget {
+  const CreditCardButton(
+      {super.key,
+      this.onPressed,
+      required this.cardData,
+      required this.config,
+      required this.onPaymentResult,
+      this.buttonTitle,
+      this.buttonStyle,
+      this.showLoading = true,
+      this.locale = const Localization.en()});
+
+  final Function(bool isLoading)? onPressed;
+  final Function onPaymentResult;
+  final PaymentConfig config;
+  final CardFormModel cardData;
+  final Localization locale;
+  final String? buttonTitle;
+  final ButtonStyle? buttonStyle;
+  final bool showLoading;
+
+  @override
+  State<CreditCardButton> createState() => _CreditCardButtonState();
+}
+
+class _CreditCardButtonState extends State<CreditCardButton> {
+  bool _isSubmitting = false;
+
+  void _saveForm() async {
```

Comment:
it looks like this function has several responsibilities and we can break it down into several smaller functions